gh-actions-workshop/presentation.md

```diff
@@ -0,0 +1,522 @@
+
+---
+
+### CI/CD pipeline tools
+
+Now that you roughly know what CI/CD is, you can try to implement it. However, trying to do it from scratch is not practical so there are many tools available on the market that try to streamline this process:
+
+- **GitHub Actions** (free for open repositories, "paid" for closed repos when you surpass the limit);
+- **GitLab Pipelines** (400 compute minutes for free, paid if you wish to use their runners);
+- **Jenkins** (open source, need to bring your infrastructure, harder to learn);
+- **CircleCI** (6000 minutes for free, need to upgrade for more)
+
+---
+
+class: inverse, center, middle
+
+
+## GitHub Actions
+### Now we're getting to the fun part ;)
+
+---
+
+### GitHub Actions - Syntax
+
+Every _workflow_ (equivalent to a pipeline) file is based on the YAML configuration syntax (indentation is required, just like Python). _Workflow_ files need to be in the `.github/workflows` folder to work. 
+
+```yaml
+name: 'Link Checker: All English'
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  check-links:
+    runs-on: ${{ fromJSON('["ubuntu-latest", "self-hosted"]')[github.repository == 'github/docs-internal'] }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 16.13.x
+          cache: npm
+
+      - name: Install
+        run: npm ci
+
+      - name: Gather files changed
+        uses: trilom/file-changes-action@a6ca26c14274c33b15e6499323aac178af06ad4b
+        with:
+          fileOutput: 'json'
+
+      - name: Show files changed
+        run: cat $HOME/files.json
+
+      - name: Link check (warnings, changed files)
+        run: |
+          ./script/rendered-content-link-checker.mjs \
+            --language en \
+            --max 100 \
+            --check-anchors \
+            --check-images \
+            --verbose \
+            --list $HOME/files.json
+
+      - name: Link check (critical, all files)
+        run: |
+          ./script/rendered-content-link-checker.mjs \
+            --language en \
+            --exit \
+            --verbose \
+            --check-images \
+            --level critical
+
+```
+
+---
+
+### GitHub Actions - Events
+
+A _workflow_ can be triggered by a plentitude of events:
+ - `push` or `pull_request` are the most common (listens for branch pushes or to PR events);
+ - `create` or `delete` listen for branch creation/deletion events;
+ - `workflow_call`, `workflow_dispatch`, and `workflow_run` can chain together _workflows_ via API or by specifying dependencies;
+ - the `schedule` can make a _workflow_ work on a recurrent basis using _cron job_ syntax. There's a [huge number of events](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) that can automate almost everything ranging from issues, comments or even automating GitHub Projects.
```

Comment:
"There's a..." be in a separate line? It appears to be a bullet of information and not a link of the all slide.